index.html

```diff
@@ -119,6 +120,24 @@ <h3>
           whether to show UI that triggers its use.
         </div>
         <section>
+          <h4>
+            <dfn>canShare()</dfn> method
+          </h4>
+          <p>
+            When the <a>canShare()</a> method is called with argument
+            <var>data</var>, run the following steps:
+          </p>
+          <ol data-link-for="ShareData">
+            <li>If none of <var>data</var>'s members <a>title</a>, <a>text</a>,
```

Comment:
This doesn't really capture what happens if `files` is not supported. I know the spec assumes that it is (since an implementation compliant with the latest spec would support file sharing), but what if it isn't?

If files isn't supported, does it return true or false if I include both `files` and some other supported field?

Reply:
Additional unsupported dictionary fields are ignored. (We could mention that explicitly.) If an implementation chose to support canShare but not file sharing, then (a) that implementation would not be consistent with any version of the spec, (b) that implementation should return false if the dictionary only contained files. If would return true if the dictionary also contained some other supported field.

Reply:
Adding text to say the method must return false if the dictionary contains `files` and file sharing is not supported would be possible for the special case of `files`, but similar text would probably not be possible for any future fields that might be added in future versions.

Reply:
> If an implementation chose to support canShare but not file sharing, then (a) that implementation would not be consistent with any version of the spec

I guess this is important for how developers are expected to use it.

Are we leaving open a possibility of an intermediate browser that supports `canShare` but not files? It's possible that a future browser will never want to support file sharing (because the OS doesn't have a concept of files, or whatever), but will want to support some other new feature that requires `canShare`. So I think we should allow this, possibly even explicitly saying that files is a MAY (not a MUST), and linking up `canShare` so it MUST return true if and only if files is accepted.

If we are allowing this possibility of `canShare` without `files`, then we need to enumerate this case, and that means developers are expected to check for both `canShare` and `files`, like this:

```js
if (navigator.share && navigator.canShare && navigator.canShare({files})) {
  await showShareButton();
  navigator.share({files});
}
```

If we aren't, then it means the very existence of `canShare` implies that `files` is supported, so then developers have a legitimate expectation of this code always being correct:

```js
if (navigator.share && navigator.canShare) {
  await showShareButton();
  navigator.share({files});
}
```

The latter option seems silly to me, so I think we should require developers to actually call `canShare` if they want to check if file sharing is allowed, which means we need to allow `canShare` to exist, but return false if given `files`.